src/uwtools/api/file.py

```diff
@@ -56,21 +59,22 @@ def link(
     """
     Link files.
 
-    :param target_dir: Path to target directory
     :param config: YAML-file path, or ``dict`` (read ``stdin`` if missing or ``None``).
+    :param target_dir: Path to target directory.
     :param cycle: A datetime object to make available for use in the config.
     :param leadtime: A timedelta object to make available for use in the config.
-    :param keys: YAML keys leading to file dst/src block
-    :param dry_run: Do not link files
+    :param keys: YAML keys leading to file dst/src block.
+    :param dry_run: Do not link files.
     :param stdin_ok: OK to read from ``stdin``?
-    :return: ``True`` if no exception is raised
+    :return: ``True`` if all links were created.
     """
-    _FileLinker(
-        target_dir=Path(target_dir),
+    linker = _FileLinker(
+        target_dir=Path(target_dir) if target_dir else None,
         config=_ensure_data_source(_str2path(config), stdin_ok),
         cycle=cycle,
         leadtime=leadtime,
         keys=keys,
         dry_run=dry_run,
-    ).go()
-    return True
+    )
+    assets: list[Asset] = linker.go()  # type: ignore
+    return all(asset.ready() for asset in assets)
```

Comment:
Updated similarly to `copy()`: The return value now indicates whether the links were all created.